docs/sources/flow/concepts/clustering.md

```diff
@@ -0,0 +1,243 @@
+of its peers and the details page of the each node's `prometheus.scrape`
+component will point to which target is being scraped by that node in its
+Debug Info.
+
+## Helm chart clustering example
+
+The easiest way to deploy an agent cluster is by making use of our
+[Helm chart][]. Here's an example of how to achieve that.
+
+The following `values.yaml` file deploys anStatefulSet for metrics
+collection. It makes use of a [headless service][] to retrieve the IPs of the
+agent pods for the `--cluster.join-addresses` argument, as well as an
+[Horizontal Pod Autoscaler][] (HPA) for dynamically matching demand.
+
+```yaml
+--- clustering-values.yaml ---
+agent:
+  mode: 'flow'
+  configMap:
+    # -- Create a new ConfigMap for the config file.
+    create: true
+    # -- Content to assign to the new ConfigMap. This is passed into `tpl` allowing for templating from values.
+    content: ''
+
+  # -- Address and port to listen for traffic on.
+  # 0.0.0.0 exposes the HTTP server to other containers.
+  listenAddr: 0.0.0.0
+  listenPort: 80
+
+  # -- Extra args to pass to `agent run`: https://grafana.com/docs/agent/latest/flow/reference/cli/run/
+  extraArgs:
+    - "--cluster.enabled"
+    - "--cluster.join-addresses=grafana-agent"  # Uses the headless service name, which defaults to the installation name.
+
+  # -- The minimum resources required for scheduling; required for using autoscaling.
+  resources:
+    requests:
+      cpu: 100m
+      memory: 200Mi
+
+controller:
+  type: 'statefulset'
+
+  # -- Whether to enable automatic deletion of stale PVCs due to a scale down operation, when controller.type is 'statefulset'.
+  enableStatefulSetAutoDeletePVC: true
+
+  autoscaling:
+    # -- Creates a HorizontalPodAutoscaler for controller type deployment.
+    enabled: true
+    # -- The lower limit for the number of replicas to which the autoscaler can scale down.
+    minReplicas: 3
+    # -- The upper limit for the number of replicas to which the autoscaler can scale up.
+    maxReplicas: 8
+    # -- Average Memory utilization across all relevant pods, a percentage of the requested value of the resource for the pods. Setting `targetMemoryUtilizationPercentage` to 0 disables Memory scaling.
+    targetMemoryUtilizationPercentage: 80
+
+service:
+  # -- Creates a Headless Service for the controller's pods.
+  enabled: true
+  type: ClusterIP
+  clusterIP: 'None'
+```
+
+Also, here's a simple River config file
+```river
+--- clustering.river ---
+logging {
+	level  = "info"
+	format = "logfmt"
+}
+
+discovery.kubernetes "pods" {
+	role = "pod"
+}
+
+prometheus.scrape "pods" {
+	clustering {
+		enabled = true
+	}
+	targets    = discovery.kubernetes.pods.targets
+	forward_to = [prometheus.remote_write.default.receiver]
+}
+
+prometheus.remote_write "default"{
+	endpoint {
+		url = env("PROMETHEUS_URL")
+		basic_auth {
+			username = env("PROMETHEUS_USERNAME")
+			password = env("PROMETHEUS_API_KEY")
+		}
```

Comment:
These environment variables aren't set in the values.yaml you've instructed the user to use.